src/main/webapp/app/exercises/programming/hestia/git-diff-report/git-diff-report.component.ts

```diff
@@ -123,7 +135,14 @@ export class GitDiffReportComponent implements OnInit {
      * @param ready Whether the diff is ready to be displayed or not.
      */
     onDiffReady(path: string, ready: boolean) {
-        this.diffsReadyByPath[path] = ready;
-        this.allDiffsReady = Object.values(this.diffsReadyByPath).reduce((a, b) => a && b);
+        const index = this.diffInformationForPaths.findIndex((info) => info.path === path);
+        if (index !== -1) {
+            this.diffInformationForPaths[index].diffReady = ready;
+            this.allDiffsReady = Object.values(this.diffInformationForPaths)
+                .map((info) => info.diffReady)
+                .reduce((a, b) => a && b);
```

Comment:
```suggestion
this.allDiffsReady = Object.values(this.diffInformationForPaths)
    .every((info) => info.diffReady);
```
using this e.g. could make the code more concise here

Reply:
Added 👍